src/sentry/static/sentry/app/views/performance/trends/chart.tsx

```diff
@@ -61,31 +60,35 @@ function transformEventStats(data: EventsStatsData, seriesName?: string): Series
   ];
 }
 
-function transformEventStatsSmoothed(data: Series[], seriesName?: string): Series[] {
+function transformEventStatsSmoothed(data: Series[], seriesName?: string) {
+  let minValue = Number.MAX_SAFE_INTEGER;
+  let maxValue = 0;
   const currentData = data[0].data;
   const resultData: SeriesDataUnit[] = [];
-  resultData.push(currentData[0]);
-
-  // Start at 1 so we don't get NaN from the movingAverage
-  for (let i = 1; i < currentData.length; i++) {
-    let value;
-    if (i < AVERAGE_WINDOW) {
-      // A rougher movingAverage for these points, but this way there isn't a gap
-      value = movingAverage(currentData, i, i);
-    } else {
-      value = movingAverage(currentData, i, AVERAGE_WINDOW);
-    }
+
+  const smoothed = smoothTrend(currentData.map(({name, value}) => [Number(name), value]));
+
+  for (let i = 0; i < smoothed.length; i++) {
+    const point = smoothed[i] as any;
+    const value = point.y;
     resultData.push({
-      name: currentData[i].name,
+      name: point.x,
       value,
     });
+    minValue = isNaN(value) ? minValue : Math.min(Math.round(value), minValue);
+    maxValue = isNaN(value) ? maxValue : Math.max(Math.round(value), maxValue);
```

Comment:
You double round() here. You could do it once and reuse the result.

Reply:
The `isNaN` check happens twice as well. You can just use an `if` here.